### PR TITLE
Committee migrations

### PR DESCRIPTION
**Pull request by denalimarsh**
Legacy committee types:
- types/committee.go: committees, proposals, votes
- types/permissions.go
- types/genesis.go
- types/codec.go: `RegisterCodec()` method. Why are registration of msgs and committees dropped in legacy/v0_13/types.go?

### CODE REVIEW

File: migrate/v0_15/migrate.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+	newAppState := MigrateAppState(appStateMap)
+	v0_15Codec := app.MakeCodec()
+	marshaledNewAppState, err := v0_15Codec.MarshalJSON(newAppState)
+	if err != nil {
+		panic(err)
+	}
+	genDoc.AppState = marshaledNewAppState
+	genDoc.GenesisTime = GenesisTime
+	genDoc.ChainID = "kava-8"
+	genDoc.ConsensusParams.Block.MaxGas = 20000000
```
Comment (karzak) [inline migrate/v0_15/migrate.go:43]: This value is unchanged, can delete this line

File: migrate/v0_15/migrate.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	tmtypes "github.com/tendermint/tendermint/types"
+
+	"github.com/kava-labs/kava/app"
+	v0_14committee "github.com/kava-labs/kava/x/committee/legacy/v0_14"
+	"github.com/kava-labs/kava/x/committee/types"
+	v0_15committee "github.com/kava-labs/kava/x/committee/types"
+)
+
+var (
+	GenesisTime = time.Date(2021, 4, 8, 15, 0, 0, 0, time.UTC)
```
Comment (karzak) [inline migrate/v0_15/migrate.go:20]: Add TODO to update this

File: migrate/v0_15/migrate_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	require.Equal(t, len(oldGenState.Committees), len(newGenState.Committees))
+	for i := 0; i < len(oldGenState.Committees); i++ {
+		require.Equal(t, len(oldGenState.Committees[i].Permissions), len(newGenState.Committees[i].GetPermissions()))
+	}
+
+	oldSPCP := oldGenState.Committees[0].Permissions[0].(v0_14committee.SubParamChangePermission)
+	newSPCP := newGenState.Committees[0].GetPermissions()[0].(v0_15committee.SubParamChangePermission)
+	require.Equal(t, len(oldSPCP.AllowedParams), len(newSPCP.AllowedParams))
+	require.Equal(t, len(oldSPCP.AllowedAssetParams), len(newSPCP.AllowedAssetParams))
+	require.Equal(t, len(oldSPCP.AllowedCollateralParams), len(newSPCP.AllowedCollateralParams))
```
Comment (karzak) [inline migrate/v0_15/migrate_test.go:53]: Add check for money markets

```
require.Equal(t, len(oldSPCP.AllowedMoneyMarkets), len(newSPCP.AllowedMoneyMarkets))
```

**Review by karzak — APPROVED** [review]
Few nits, otherwise LGTM

File: x/committee/legacy/v0_14/types.go [hunk 4]
```diff
@@ -0,0 +1,999 @@
...
+	// Proposals
+	cdc.RegisterInterface((*PubProposal)(nil), nil)
+
+	// Permissions
+	cdc.RegisterInterface((*Permission)(nil), nil)
+	cdc.RegisterConcrete(GodPermission{}, "kava/GodPermission", nil)
+	cdc.RegisterConcrete(SimpleParamChangePermission{}, "kava/SimpleParamChangePermission", nil)
+	cdc.RegisterConcrete(TextPermission{}, "kava/TextPermission", nil)
+	cdc.RegisterConcrete(SoftwareUpgradePermission{}, "kava/SoftwareUpgradePermission", nil)
+	cdc.RegisterConcrete(SubParamChangePermission{}, "kava/SubParamChangePermission", nil)
```
Comment (karzak) [inline x/committee/legacy/v0_14/types.go:998]: Doesn't matter for migrations, but I think we should register the messages here, just for completeness.

File: migrate/v0_15/migrate.go (hunk 1 above)
Comment (denalimarsh) [reply migrate/v0_15/migrate.go]: Removed in https://github.com/Kava-Labs/kava/pull/909/commits/156e6deb9be9f1aaad34e1ae37f8d614b6092fdd

File: migrate/v0_15/migrate.go (hunk 2 above)
Comment (denalimarsh) [reply migrate/v0_15/migrate.go:21]: Added in https://github.com/Kava-Labs/kava/pull/909/commits/156e6deb9be9f1aaad34e1ae37f8d614b6092fdd

File: migrate/v0_15/migrate_test.go (hunk 3 above)
Comment (denalimarsh) [reply migrate/v0_15/migrate_test.go:53]: Addressed in https://github.com/Kava-Labs/kava/pull/909/commits/156e6deb9be9f1aaad34e1ae37f8d614b6092fdd

File: x/committee/legacy/v0_14/types.go (hunk 4 above)
Comment (denalimarsh) [reply x/committee/legacy/v0_14/types.go:1008]: Registered msgs on the legacy codec in https://github.com/Kava-Labs/kava/pull/909/commits/bea0a6d9a6957cbcdacf13cccb3ecaed37ef5d91

**Pull request merged**